apps/shipments/management/commands/replicate_shipments.py

```diff
@@ -0,0 +1,97 @@
+import logging
+import sys
+import time
+
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from shipchain_common.exceptions import RPCError
+
+from apps.eth.models import EthAction
+from apps.eth.rpc import EventRPCClient
+from apps.jobs.models import AsyncJob, Message
+from apps.jobs.tasks import AsyncTask
+from apps.shipments.models import Shipment
+
+logger = logging.getLogger('transmission')
+logger.setLevel(settings.LOG_LEVEL)
+
+
+class Command(BaseCommand):
+    help = 'Replicate shipment related transactions to the sidechain.'
+    rpc_client = None
+    successfull_shipments = []
+    unsuccessfull_shipments = []
+    async_job = None
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--shipment_id',
+            type=str,
+            default=None,
+            help='Replicate specific shipment instead of all of them.',
+        )
+
+    def _replicate_shipment(self, shipment):
+        self.successfull_shipments.append(shipment.id)
+        for async_job in AsyncJob.objects.filter(shipment=shipment):
+            try:
+                eth_action = EthAction.objects.filter(async_job=async_job).first()
+                if eth_action:
+                    logger.info(f'Deleting Eth Action: {eth_action.transaction_hash}')
+                    eth_action.delete()
+                AsyncTask(async_job.id).rerun()
+                self.async_job = async_job
+
+            # pylint:disable=broad-except
+            except Exception as exc:
+                if shipment.id in self.successfull_shipments:
+                    self.successfull_shipments.remove(shipment.id)
+                    self.unsuccessfull_shipments.append(shipment.id)
+                logger.error(exc)
+
+    def _unsubscribe(self):
+        self.rpc_client = EventRPCClient()
+        try:
+            self.rpc_client.unsubscribe("LOAD", settings.LOAD_VERSION)
+        except RPCError as exc:
+            if 'Error: EventSubscription not found' not in exc.detail:
+                raise exc
+
+    def _resubscribe(self):
+        if not self.async_job:
+            logger.warning('Unable to find last blockheight to subscribe to. '
+                           'Need to manually subscribe to latest blockheight')
+            return
+
+        message = Message.objects.filter(async_job=self.async_job).last()
+        if "blockNumber" not in message.body:
+            time.sleep(15)
+            message = Message.objects.filter(async_job=self.async_job).last()
+        if "blockNumber" in message.body:
+            self.rpc_client.subscribe("LOAD", settings.LOAD_VERSION, last_block=message.body["blockNumber"])
+        else:
+            logger.warning('Unable to find last blockheight to subscribe to. '
+                           'Need to manually subscribe to latest blockheight')
+
+    def handle(self, *args, **options):
+        self._unsubscribe()
```

Comment:
What happens if shipments are created/updated via api during this migration? Should we consider a maintenance page/return 503 for all endpoints during the migration?

Reply:
If a new shipment is created during this migration, those transactions will already be sending to the new chain and there is nothing to replicate.  If we solve the "filter only unmigrated shipments" question you asked then we shouldn't need to be concerned with new shipments being created during this process.

Reply:
Would there be any concerns at all about the difference in the last block when resubscribing if a shipment was created while a shipment is rerunning?